Source/SwiftLintFramework/Rules/OperatorUsageWhitespaceRule.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  OperatorUsageWhitespaceRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 13/12/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct OperatorUsageWhitespaceRule: OptInRule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "operator_usage_whitespace",
+        name: "Operator Usage Whitespace",
+        description: "Operators should be surrounded by a single whitespace " +
+                     "when they are being used.",
+        nonTriggeringExamples: [
+            "let foo = 1 + 2\n",
+            "let foo = 1 > 2\n",
+            "let foo = !false\n",
+            "let foo: Int?\n",
+            "let foo: Array<String>\n",
+            "let foo: [String]\n",
+            "let foo = 1 + \n  2\n",
+            "let range = 1...3\n",
+            "let range = 1 ... 3\n",
+            "let range = 1..<3\n"
+        ],
+        triggeringExamples: [
+            "let foo = 1+2\n",
+            "let foo = 1   + 2\n",
+            "let foo = 1   +    2\n",
+            "let foo = 1 +    2\n",
+            "let foo=1+2\n",
+            "let foo=1 + 2\n",
+            "let foo=bar\n",
+            "let range = 1 ..<  3\n"
+        ]
+    )
+
+    public func validateFile(_ file: File) -> [StyleViolation] {
+        let escapedOperators = ["/", "=", "-", "+", "*", "|", "^", "~"]
+            .map({ "\\\($0)" }).joined()
+        let rangePattern = "\\.\\.(\\.|<)" // ... or ..<
+        let operators = "([\(escapedOperators)%<>&]+|\(rangePattern))"
+
+        let oneSpace = "[^\\S\\r\\n]" // to allow lines ending with operators to be valid
+        let zeroSpaces = oneSpace + "{0}"
+        let manySpaces = oneSpace + "{2,}"
+        let variableOrNumber = "(\(RegexHelpers.varName)|\(RegexHelpers.number))"
+
+        let pattern1 = variableOrNumber + zeroSpaces + operators + zeroSpaces + variableOrNumber
```

Comment:
To make this a little more DRY, perhaps something along the lines of:

```swift
let spaces = [zeroSpaces, oneSpaces, twoSpaces]

let patterns = Set(spaces.map { firstSpace in
  return spaces.map { secondSpace in
    return variableOrNumber + firstPace + operators + secondSpace + variableOrNumber
 }
}.flatmap { $0 })
```

🤔 Still non-trivial, but I haven't been able to come up with a better pattern.